### PR TITLE
Feat/android max demo app migration

### PR DESCRIPTION
**Pull request by richashukla23**

### CODE REVIEW

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java [hunk 1]
```diff
@@ -67,15 +79,77 @@ public void onSdkInitialized(final AppLovinSdkConfiguration config)
...
+                    .setNeutralButton( "OK", null )
+                    .show();
+        }
+    }
+
+    private Intent getContactIntent()
+    {
+        Intent intent = new Intent( Intent.ACTION_SENDTO );
+        intent.setType( "text/plain" );
+        intent.setData( Uri.parse( "mailto:" + "support@applovin.com" ) );
```
Comment (santoshbagadi) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:100]: From [Android's documentation](https://developer.android.com/guide/components/intents-common#Email), it looks like you would have to set the email using the [`Intent.EXTRA_EMAIL`](https://developer.android.com/reference/android/content/Intent#EXTRA_EMAIL).

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/SplashActivity.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package com.applovin.enterprise.apps.demoapp.ads.applovin;
```
Comment (santoshbagadi) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/SplashActivity.java:1]: This Activity is redundant to the one [here](https://github.com/AppLovin/AppLovin-MAX-SDK-Android/blob/master/AppLovin%20MAX%20Demo%20App%20-%20Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/SplashActivity.java) . Please remove.

File: AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.kt [hunk 3]
```diff
@@ -21,13 +35,24 @@ import com.applovin.sdk.AppLovinMediationProvider
...
 import kotlinx.android.synthetic.main.content_main.*
+import kotlinx.android.synthetic.main.activity_list.*
 
 class MainActivity : AppCompatActivity(),
         MainRecyclerViewAdapter.OnMainListItemClickListener
 {
+
+    private lateinit var muteToggleMenuItem: MenuItem
+
     private val mainListItems: List<ListItem> by lazy {
```
Comment (santoshbagadi) [inline AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.kt:46]: Missing `SUPPORT` section here.

Comment (richashukla23) [reply AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.kt:46]: Added:

SectionHeader("SUPPORT"),
                AdType("Resources", Intent(Intent.ACTION_VIEW, Uri.parse("https://support.applovin.com/support/home"))),
                AdType("Contact", Intent(makeContactIntent()))

File: AppLovin MAX Demo App - Java/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -29,20 +29,84 @@
...
+            android:name="com.applovin.enterprise.apps.demoapp.ads.max.RewardedAdActivity"
             android:configChanges="orientation|screenSize|keyboardHidden"
             android:parentActivityName="com.applovin.enterprise.apps.demoapp.MainActivity" />
         <activity
-            android:name="com.applovin.enterprise.apps.demoapp.ads.InterstitialAdActivity"
+            android:name="com.applovin.enterprise.apps.demoapp.ads.max.InterstitialAdActivity"
             android:configChanges="orientation|screenSize|keyboardHidden"
             android:parentActivityName="com.applovin.enterprise.apps.demoapp.MainActivity" />
+        <activity
+            android:name=".ads.applovin.interstitials.InterstitialDemoMenuActivity"
```
[the remark this answers is not in the document]
Comment (richashukla23) [reply AppLovin MAX Demo App - Java/app/src/main/AndroidManifest.xml:48]: Updated full package names.

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java [hunk 5]
```diff
@@ -67,15 +79,77 @@ public void onSdkInitialized(final AppLovinSdkConfiguration config)
...
+        items.add( new AdType( "MRECs", new Intent( this, MRecDemoMenuActivity.class ) ) );
+        items.add( new AdType( "Event Tracking", new Intent( this, EventTrackingActivity.class ) ) );
+        items.add( new SectionHeader( "MAX" ) );
         items.add( new AdType( "Interstitial", new Intent( this, InterstitialAdActivity.class ) ) );
         items.add( new AdType( "Rewarded", new Intent( this, RewardedAdActivity.class ) ) );
         items.add( new AdType( "Programmatic Banners", new Intent( this, ProgrammaticBannerAdActivity.class ) ) );
         items.add( new AdType( "Layout Editor Banners", new Intent( this, LayoutEditorBannerAdActivity.class ) ) );
+        items.add( new SectionHeader( "SUPPORT" ) );
+        items.add( new AdType( "Resources", new Intent( Intent.ACTION_VIEW, Uri.parse( "https://support.applovin.com/support/home" ) ) ));
+        //items.add( new AdType( "AL Contact", new Intent( this, getContactIntent() ) ) );
```
[the remark this answers is not in the document]
Comment (richashukla23) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:152]: Uncommented and fixed.

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java [hunk 6]
```diff
@@ -67,15 +79,77 @@ public void onSdkInitialized(final AppLovinSdkConfiguration config)
...
+        {
+            return getResources().getDrawable( drawableId, getTheme() );
+        }
+        else
+        {
+            return getResources().getDrawable( drawableId );
+        }
+    }
+
+
```
[the remark this answers is not in the document]
Comment (richashukla23) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:133]: Removed extra space.

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/DemoMenuActivity.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        listView = (ListView) findViewById( R.id.listView );
+
+        setupListViewFooter();
+        setupListViewContents( getListViewContents() );
+    }
+
+    protected void setupListViewFooter()
+    {
+
+    }
```
[the remark this answers is not in the document]
Comment (richashukla23) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/DemoMenuActivity.java:39]: I removed the empty method and the method call. 

Note - empty method and call is present in AppLovin Demo DemoMenuActivity too.

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/SplashActivity.java (hunk 2 above)
Comment (richashukla23) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/SplashActivity.java:1]: @santoshbagadi Can you confirm if this means - remove SplashActivity altogether for max demo app?

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java [hunk 8]
```diff
@@ -108,4 +182,6 @@ public boolean onOptionsItemSelected(final MenuItem item)
 
         return super.onOptionsItemSelected( item );
     }
+
+
```
[the remark this answers is not in the document]
Comment (richashukla23) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:186]: Remove extra space.

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/SplashActivity.java (hunk 2 above)
Comment (santoshbagadi) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/SplashActivity.java:1]: No, just remove the duplicate one please. Keep the existing one.

**Comment by santoshbagadi** [comment]
Also, no need to reply to each comment. For simple changes, you can just mark the comments resolved.

Just comment on the overall PR when it's ready for review again.

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java (hunk 1 above)
Comment (richashukla23) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:100]: Updated:

intent.putExtra(Intent.EXTRA_EMAIL, "support@applovin.com");

**Comment by thomasmso** [comment]
Also @richashukla23 - no need to add the Asana ticket to each of your commit, only need to do it for your PR.

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/interstitials/InterstitialDemoMenuActivity.java [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+import com.applovin.enterprise.apps.demoapp.R;
+
+public class InterstitialDemoMenuActivity
+        extends DemoMenuActivity
+{
+    @Override
+    protected DemoMenuItem[] getListViewContents()
+    {
+        DemoMenuItem[] result = {
+                new DemoMenuItem( "Basic Integration", "Quick interstitial ads integration", new Intent( this, InterstitialBasicIntegrationActivity.class ) ),
```
Comment (thomasmso) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/interstitials/InterstitialDemoMenuActivity.java:29]: Can we remove all subtitles from the demo app? The iOS demo app doesn't have subtitles for the rows anymore. Makes for cleaner/simpler design. (Same for the other sections).

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/interstitials/InterstitialDemoMenuActivity.java [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+                new DemoMenuItem( "Manually loading ad", "Use this for greater control over the ad load process", new Intent( this, InterstitialManualLoadingActivity.class ) ),
+                new DemoMenuItem( "Zone Integration", "Create different user experiences of the same ad type", new Intent( this, InterstitialZoneActivity.class ) )
+        };
+        return result;
+    }
+
+    /**
```
Comment (thomasmso) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/interstitials/InterstitialDemoMenuActivity.java:36]: Why do we have the `InterstitialManualLoadingActivity` class here? It already exists in another file.

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java [hunk 11]
```diff
@@ -69,17 +79,76 @@ public void onSdkInitialized(final AppLovinSdkConfiguration config)
         } );
     }
 
+    private void checkSdkKey()
+    {
+        final String sdkKey = AppLovinSdk.getInstance( getApplicationContext() ).getSdkKey();
+        if ( "YOUR_SDK_KEY".equalsIgnoreCase( sdkKey ) )
+        {
+            new AlertDialog.Builder( this )
```
Comment (tiandrew) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:87]: We have a utility method called `showOkAlertDialog()` in `TestAppUtils.java`, we can also add `setCancelable( false )` to it

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java [hunk 12]
```diff
@@ -69,17 +79,76 @@ public void onSdkInitialized(final AppLovinSdkConfiguration config)
...
+                    .setNeutralButton( "OK", null )
+                    .show();
+        }
+    }
+
+    private Intent getContactIntent()
+    {
+        Intent intent = new Intent( Intent.ACTION_SENDTO );
+        intent.setType( "text/plain" );;
+        intent.putExtra(Intent.EXTRA_EMAIL, "support@applovin.com");
```
Comment (tiandrew) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:100]:
```suggestion
        intent.putExtra( Intent.EXTRA_EMAIL, "support@applovin.com" );
```

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java (hunk 11 above)
Comment (richashukla23) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:87]: @stnaqvi I noticed that checkSdkKey() method is never used. Do you think we should still go ahead and make the above update within checkSdkKey() ?

Comment (richashukla23) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:87]: I think it is supposed to be called somewhere within MainActivity, I will look into it.

Comment (thomasmso) [reply AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.java:87]: Hmm I don't think we have it in the new demo app but I'm down to add it in similar fashion to what we have [here](https://github.com/AppLovin/Android-SDK-Demo/blob/6408e9d6122b782b161f540267bfcfb9a2735a92/Android-Demo-App/src/main/java/com/applovin/apps/demoapp/MainActivity.java#L56).

**Pull request merged**